### ISSUE TITLE
Add value-slash-space-*

### ISSUE THREAD

**Issue by in-in** [stylelint/stylelint#4657]
> What is the problem you're trying to solve?

I want to remove inconsistent values for a grid properties e.g. `grid-row: 1/span 6` and `grid-row: 1 / span 6`. Both examples are valid css syntax

> What solution would you like to see?

A new rule(s) that allows or disallows a space before and after a slash. In my case, I only need one rule that disallow an unspaced slash as the rule [function-calc-no-unspaced-operator](https://stylelint.io/user-guide/rules/function-calc-no-unspaced-operator) does.

**Comment by jeddy3** [comment]
@in-in Thanks for the request and for using the template.

I agree that we should support this. There is a lot to consider, though. 

> In my case, I only need one rule that disallow an unspaced slash as the rule `function-calc-no-unspaced-operator` does.

The majority of [the whitespace rules](https://stylelint.io/user-guide/rules/about#whitespace-rules) come in sets of four (for example, [`value-list-comma-*-*`](https://stylelint.io/user-guide/rules/list#value-list)), which [can be combined together](https://stylelint.io/user-guide/rules/combine#-newlinespace-before-and--newlinespace-after-rules) to enforce a variety of conventions.

The `function-calc-no-unspaced-operator` was an exception because [the `*` and `/` operators must have spaces to be valid CSS](https://github.com/stylelint/stylelint/pull/116#issuecomment-105239110). The rule enforces space around the `+` and `-` operators to be consistent.

I suggest we create a set of four rules:

- `value-slash-space-before`
- `value-slash-space-after`
- `value-slash-newline-before`
- `value-slash-newline-after`

Where:

- `value` is the scope
- `slash` is the punctuation

Because:

- the scope will work in conjunction with the `value-list-comma-*-*` rules
- "slash" is used in both [the grid](https://www.w3.org/TR/css-grid-1/#placement-shorthands) and [color specs](https://drafts.csswg.org/css-color/#color-function)

All the rules will ignore the division operator (`/`) within `calc` functions.

Conventions we'll want to support include:

```css
a { grid-row: 1/span 6 }
```

```css
a { grid-row: 1 / span 6 }
```

```css
a { 
  grid-row: 1 /
    span 6
}
```

```css
a { 
  grid-row: 1
    / span 6
}
```

[Named areas](https://www.w3.org/TR/css-grid-1/#grid-template-areas-property) will need to be considered to:

```css
main {
  grid: "H    H "
        "A    B "
        "F    F " 30px
  /     auto 1fr;
}
```

I suggest the rules check slashes within functions (that aren't `calc`). For example, for [slash separated opacity](https://drafts.csswg.org/css-color/#changes-from-3):

```css
a { color: rgba(0 0 0 / 30%); }
```

We can always move this functionality to a dedicated a set of `function-slash-*-*` rules if it proves problematic.

There's a fair bit to mull over, so I suggest we add just the rules and options needed to enforce your use case to begin with:

- `value-slash-space-before`: `"always-single-line"`
- `value-slash-space-after`: `"always-single-line"`

This will allow you to enforce:

```css
a { grid-row: 1 / span 6 }
```

While not limiting us when we come to support more stylistics conventions in the future.

Please shout up if anyone sees any problems with this approach. The grid specification is significant, and I've likely overlooked some potential use cases involving slashes. 

In the meantime, I'll label this up as ready to implement as starting on the rules is an excellent way to uncover issues.

Rule one:

- Name: `value-slash-space-after`
- Primary option: `"always-single-line"`
- Secondary options: None
- Message: Expected single space after "/"'
- Description: "Require a single space after the slashes of values."
- Section: "Stylistic issues" -> "Value"

Rule two:

- Name: `value-slash-space-before`
- Primary option: `"always-single-line"`
- Secondary options: None
- Message: Expected single space before "/"'
- Description: "Require a single space before the slashes of values."
- Section: "Stylistic issues" -> "Value"

@in-in Please consider [contributing](https://stylelint.io/contributing) these rules if you have time. You'll likely want to use the equivalent `value-list-comma-*-*` rules as blueprints.

There are [steps on how to add a new rule](https://stylelint.io/developer-guide/rules#add-a-rule) in the Developer guide.

**Comment by jeddy3** [comment]
Closing [as stylistic rules are frozen](https://github.com/stylelint/stylelint/issues/5674). The community is welcome to create the rules [as plugins](https://stylelint.io/developer-guide/plugins).
